### PR TITLE
Upgrade release-docker workflow

### PR DESCRIPTION
**Pull request by underyx**
PR checklist:

- [ ] Documentation is up-to-date
- [ ] Changelog is up-to-date
- [ ] Change has no security implications (otherwise, ping security team)

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -87,23 +87,35 @@ jobs:
...
 
   release-docker:
-    name: Build and Push Semgrep Docker Container
+    name: Build and push Semgrep Docker image
     runs-on: ubuntu-latest
     if: ${{ !contains(github.ref,'-test-release') }}
     steps:
-      - uses: actions/checkout@v3
-        with:
-          submodules: true
```
Comment (underyx) [inline .github/workflows/release.yml:96]: Removed checkout step cause docker/build-push-action automatically uses the git repo content from the push event: https://github.com/docker/build-push-action#git-context

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -87,23 +87,35 @@ jobs:
...
           password: ${{ secrets.DOCKER_PASSWORD }}
-          repository: returntocorp/semgrep
-          tags: ${{ steps.get_version.outputs.VERSION }}, latest
+
+      - id: docker-meta
+        name: Set tags and labels
+        uses: docker/metadata-action@v3
+        with:
+          images: ${{ github.repository }}
+          tags: type=match,pattern=v(\d.\d.\d)
```
Comment (underyx) [inline .github/workflows/release.yml]: docs here: https://github.com/docker/metadata-action#typematch

Comment (underyx) [inline .github/workflows/release.yml]:
```suggestion
          tags: type=match,pattern=\d.\d.\d
```

**Review by brendongo — APPROVED** [review]

**Comment by brendongo** [comment]
Should we change the docker.yaml jobs to use the same config as this with push: False

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -87,23 +87,35 @@ jobs:
...
           password: ${{ secrets.DOCKER_PASSWORD }}
-          repository: returntocorp/semgrep
-          tags: ${{ steps.get_version.outputs.VERSION }}, latest
+
+      - id: docker-meta
+        name: Set tags and labels
+        uses: docker/metadata-action@v3
+        with:
+          images: ${{ github.repository }}
+          tags: type=match,pattern=\d.\d.\d
```
Comment (underyx) [inline .github/workflows/release.yml]:
```suggestion
  tags: |
    type=semver,pattern={{version}}
    type=semver,pattern={{major}}.{{minor}}
```

File: .github/workflows/release.yml [hunk 4]
```diff
@@ -87,23 +87,37 @@ jobs:
...
-          tags: ${{ steps.get_version.outputs.VERSION }}, latest
+
+      - id: docker-meta
+        name: Set tags and labels
+        uses: docker/metadata-action@v3
+        with:
+          images: ${{ github.repository }}
+  tags: |
+    type=semver,pattern={{version}}
+    type=semver,pattern={{major}}.{{minor}}
```
Comment (underyx) [inline .github/workflows/release.yml]:
```suggestion
          type=semver,pattern={{version}}
          type=semver,pattern={{major}}.{{minor}}
```

File: .github/workflows/release.yml [hunk 5]
```diff
@@ -87,23 +87,37 @@ jobs:
...
+
+      - id: docker-meta
+        name: Set tags and labels
+        uses: docker/metadata-action@v3
+        with:
+          images: ${{ github.repository }}
+  tags: |
+          type=semver,pattern={{version}}
+          type=semver,pattern={{major}}.{{minor}}
+          # when running on a tag push, :latest is added automatically
```
Comment (underyx) [inline .github/workflows/release.yml]:
```suggestion
        tags: |
          type=semver,pattern={{version}}
          type=semver,pattern={{major}}.{{minor}}
          # when running on a tag push, :latest is added automatically
```

File: .github/workflows/release.yml [hunk 6]
```diff
@@ -87,23 +87,37 @@ jobs:
...
+
+      - id: docker-meta
+        name: Set tags and labels
+        uses: docker/metadata-action@v3
+        with:
+          images: ${{ github.repository }}
+        tags: |
+          type=semver,pattern={{version}}
+          type=semver,pattern={{major}}.{{minor}}
+          # when running on a tag push, :latest is added automatically
```
Comment (underyx) [inline .github/workflows/release.yml]:
```suggestion
          tags: |
            type=semver,pattern={{version}}
            type=semver,pattern={{major}}.{{minor}}
            # when running on a tag push, :latest is added automatically
```

File: .github/workflows/release.yml [hunk 7]
```diff
@@ -87,23 +87,37 @@ jobs:
...
+
+      - id: docker-meta
+        name: Set tags and labels
+        uses: docker/metadata-action@v3
+        with:
+          images: ${{ github.repository }}
+          tags: |
+            type=semver,pattern={{version}}
+            type=semver,pattern={{major}}.{{minor}}
+            # when running on a tag push, :latest is added automatically
```
Comment (underyx) [inline .github/workflows/release.yml]:
```suggestion
          # :latest is also added automatically
          tags: |
            type=semver,pattern={{version}}
            type=semver,pattern={{major}}.{{minor}}
```

**Review by spencerdrak — APPROVED** [review]

**Pull request merged**